rust/agama-lib/src/localization/client.rs

```diff
@@ -0,0 +1,50 @@
+use crate::error::ServiceError;
+use crate::proxies::LocaleProxy;
+use zbus::Connection;
+
+/// D-Bus client for the software service
+pub struct LocalizationClient<'a> {
+    localization_proxy: LocaleProxy<'a>,
+}
+
+impl<'a> LocalizationClient<'a> {
+    pub async fn new(connection: Connection) -> Result<LocalizationClient<'a>, ServiceError> {
+        Ok(Self {
+            localization_proxy: LocaleProxy::new(&connection).await?,
+        })
+    }
+
+    pub async fn language(&self) -> Result<String, ServiceError> {
+        let locales = self.localization_proxy.locales().await?;
+        if locales.is_empty() {
+            // FIXME is this right?
```

Comment:
well, maybe more correct way is to use `Result<Option<String>, ServiceError>` and return `Ok(None)` here.